ktor-server/ktor-server-plugins/ktor-server-sessions/jvm/src/io/ktor/server/sessions/KotlinxSessionSerializer.kt

```diff
@@ -0,0 +1,397 @@
+/*
+ * Copyright 2014-2022 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.server.sessions
+
+import io.ktor.http.*
+import kotlinx.serialization.*
+import kotlinx.serialization.descriptors.*
+import kotlinx.serialization.encoding.*
+import kotlinx.serialization.internal.*
+import kotlinx.serialization.modules.*
+
+/**
+ * Returns a [SessionSerializer] based on `kotlinx.serialization` library.
+ */
+public inline fun <reified T : Any> kotlinxSessionSerializer(
+    format: StringFormat
+): KotlinxSessionSerializer<T> {
+    return kotlinxSessionSerializer(serializer(), format)
+}
+
+/**
+ * Returns a [SessionSerializer] based on `kotlinx.serialization` library.
+ */
+public fun <T : Any> kotlinxSessionSerializer(
+    serializer: KSerializer<T>,
+    format: StringFormat
+): KotlinxSessionSerializer<T> {
+    return KotlinxSessionSerializer(format, serializer)
+}
+
+/**
+ * Implementation of [SessionSerializer] based on `kotlinx.serialization` library.
+ */
+public class KotlinxSessionSerializer<T : Any> internal constructor(
```

Comment:
BTW, why `KotlinxSessionSerializer` and `SessionsBackwardCompatibleFormat` as classes are public?
isn't it better for reducing public API to:
1. make `fun KotlinxSessionSerializer(...)` to just return `SessionSerializer`
2. create `fun SessionsBackwardCompatibleFormat(module): StringFormat` or even just `fun KotlinxBackwardCompatibleSessionSerializer(...): SessionSerializer`

Reply:
Yes, good point. I hide everything behind functions.